src/FailureDetectorConfig.md

```diff
@@ -85,3 +88,59 @@ config.setProperty("hazelcast.heartbeat.phiaccrual.failuredetector.sample.size",
+- `hazelcast.icmp.timeout` (default 500) - Number of milliseconds until a ping attempt is considered failed if there was no reply.
+- `hazelcast.icmp.max.attempts` (default 3) - The maximum number of ping attempts before the member/node gets suspected by the detector.
+- `hazelcast.icmp.interval` (default 1000) - The interval, in millis, between each ping attempt.
+- `hazelcast.icmp.ttl` (default 0) - The maximum number of hops the packets should go through or 0 for the default
+
+In the above configuration, the Ping detector will attempt 3 pings, one every second, and will wait up-to 500ms for each to complete. If after 3 seconds,
+there was no successful ping, the member will get suspected.
+
+To enforce the [Requirements](#Requirements) the property `hazelcast.icmp.echo.fail.fast.on.startup` can also be set to `true`, in which case, if any of the requirements
+isn't met, Hazelcast will fail to start.
+
+Below a summary table of all possible combinations of the ping failure detector.
+
+| ICMP  	| Parallel 	| Fail-Fast 	| Description                                                                                                                                                                                                                 	| Linux                                                                                                  	| Windows                       	| macOS                                                                	|
+|-------	|----------	|-----------	|-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------	|--------------------------------------------------------------------------------------------------------	|-------------------------------	|----------------------------------------------------------------------	|
+| false 	| false    	| false     	| Completely disabled                                                                                                                                                                                                         	| N/A                                                                                                    	| N/A                           	| N/A                                                                  	|
+| true  	| false    	| false     	| Legacy ping mode. Pings nodes up-to 5 times after a pre-defined no heartbeat period has passed. If all 5 attempts fail, node gets suspected. It relies on the configured failure detector (OSI Layer 7) before it kicks-in. 	| Supported  ICMP Echo if available - Falls back on TCP Echo on port 7                                   	| Supported  TCP Echo on port 7 	|  Supported ICMP Echo if available - Falls back on TCP Echo on port 7 	|
```

Comment:
I still get confused a bit with the sentence. I wrote and read it several time and what helped me the most to understand was to clarify the "it"s. I suggest changing the last sentence to:
"The failure relies on the configured failure detector (OSI Layer 7) before ICMP failure detector kicks-in."

Reply:
Revisited